src/app/(app)/(tenants)/dashboard/page.tsx

```
@@ -1,10 +1,38 @@
+}: DashboardPageProps) {
+  const searchParams = await searchParamsPromise;
+
+  // Guard dashboard behind auth so email deep-links prompt sign-in if needed.
+  const session = await caller.auth.session();
+  if (!session.user) {
+    // Preserve query params (e.g., tenant context) so links return correctly.
+    const params = new URLSearchParams();
+    if (searchParams) {
+      for (const [key, value] of Object.entries(searchParams)) {
+        if (Array.isArray(value)) {
+          value.forEach((entry) => params.append(key, entry));
```

Biome (2.3.14)

[error] This callback passed to forEach() iterable method should not return a value.

Either remove this return or remove the returned value.


(lint/suspicious/useIterableCallbackReturn)